### ISSUE TITLE
Parse species tree to produce population configurations and demographic models.

### ISSUE THREAD

**Issue by jeromekelleher** [tskit-dev/msprime#891]
A useful feature suggested by @mmatschiner would be to parse the output of these programs like StarBEAST to produce the simulation descriptions that can be run in msprime.

An example of such an interface would be
```python
def parse_species_tree(tree, format="newick"):
     """
     Parse the specified species tree description in the specified format, and return a tuple 
     (population_configurations, demographic_events).
     """
```

It's probably a good idea to put this in its own module, so let's create a new file ``species_tree.py`` 
and corresponding test module ``test_species_tree.py``. For tests, we'd want to 

- get some basic test coverage on all the code
- Put in a few small examples in each of the supported formats, and verify that we get the correct result.
- Verify that we do sensible things when presented with malformed input.

How does this sound @mmatschiner?

**Comment by mmatschiner** [comment]
Sounds good. I assume the two files should go into msprime/msprime and msprime/tests, respectively.

**Comment by jeromekelleher** [comment]
Yes, exactly.

**Comment by andrewkern** [comment]
Seems like the idea here is to go from a Newick tree to population split model, yes? does the population_configuration follow naturally from that? it's early, and maybe i'm being a bit dense, but it seems like that info (which samples were from which populations) wouldn't be in a species tree?

**Comment by mmatschiner** [comment]
Yes, that's the idea. You're right that the species tree doesn't contain any information about samples per species. The function is currently this:
```
def parse_species_tree(
        filename=None,
        branch_length_units="gen",
        sample_size=None,
        Ne=None,
        generation_time=None):
```
So the specified sample size applies to all populations. Unless the population size is encoded in the tree (as it is for trees written by StarBEAST), the specified Ne also applies to all populations. generation_time is required when the units of branch lengths are not generations, but years or millions of years.

**Comment by andrewkern** [comment]
got it. one design idea then-- maybe sample size in `parse_species_tree()` should take a tuple as input (if it doesn't already) to drive home to the user that they will be setting this in each population?

**Comment by mmatschiner** [comment]
I see. I guess I could let users choose between something like `sample_size_each_pop` (accepting a number which then applies to all pops) and `sample_sizes_per_pop` (accepting a tuple with sample sizes specified individually for each pop). Not sure, though, what would be the best way for the user to connect sample sizes and populations:

1. - providing a dictionary instead of a tuple
2. - providing a tuple in which population sizes are ordered according to the alphabetical order of species names
3. - providing an input file with a table with two columns for species name and sample size.

**Comment by jeromekelleher** [comment]
I don't think we need the ``sample_size`` option here, at least initially. The simulate function takes a ``samples`` argument which is fully flexible, and the user can use that in conjunction with the population configurations and demographic events returned by this function. (Or, they could set the ``sample_size`` attribute themselves on the returned PopulationConfiguration objects.)

**Comment by mmatschiner** [comment]
Even better. I didn't know about the `samples` argument.

**Comment by jeromekelleher** [comment]
Closed in #929
